libraries/AP_Quicktune/AP_Quicktune.cpp

```diff
@@ -0,0 +1,672 @@
+    } else if (param == param_s::RLL_FF || param == param_s::PIT_FF || param == param_s::YAW_FF){
+        return stages::FF;
+    } else {
+        return stages::END;
+    }
+}
+
+AP_Float *AP_Quicktune::get_param_pointer(AP_Quicktune::param_s param)
+{
+    switch (param)
+    {
+        case param_s::RLL_P:
+            return &attitude_control.get_rate_roll_pid().kP();
+        case param_s::RLL_I:
+            return &attitude_control.get_rate_roll_pid().kI();
+        case param_s::RLL_D:
+            return &attitude_control.get_rate_roll_pid().kD();
+        case param_s::RLL_SMAX:
+            return &attitude_control.get_rate_roll_pid().slew_limit();
+        case param_s::RLL_FLTT:
+            return &attitude_control.get_rate_roll_pid().filt_T_hz();
+        case param_s::RLL_FLTD:
+            return &attitude_control.get_rate_roll_pid().filt_D_hz();
+        case param_s::RLL_FLTE:
+            return &attitude_control.get_rate_roll_pid().filt_E_hz();
+        case param_s::RLL_FF:
+            return &attitude_control.get_rate_roll_pid().ff();
+        case param_s::PIT_P:
+            return &attitude_control.get_rate_pitch_pid().kP();
+        case param_s::PIT_I:
+            return &attitude_control.get_rate_pitch_pid().kI();
+        case param_s::PIT_D:
+            return &attitude_control.get_rate_pitch_pid().kD();
+        case param_s::PIT_SMAX:
+            return &attitude_control.get_rate_pitch_pid().slew_limit();
+        case param_s::PIT_FLTT:
+            return &attitude_control.get_rate_pitch_pid().filt_T_hz();
+        case param_s::PIT_FLTD:
+            return &attitude_control.get_rate_pitch_pid().filt_D_hz();
+        case param_s::PIT_FLTE:
+            return &attitude_control.get_rate_pitch_pid().filt_E_hz();
+        case param_s::PIT_FF:
+            return &attitude_control.get_rate_pitch_pid().ff();
+        case param_s::YAW_P:
+            return &attitude_control.get_rate_yaw_pid().kP();
+        case param_s::YAW_I:
+            return &attitude_control.get_rate_yaw_pid().kI();
+        case param_s::YAW_D:
+            return &attitude_control.get_rate_yaw_pid().kD();
+        case param_s::YAW_SMAX:
+            return &attitude_control.get_rate_yaw_pid().slew_limit();    
+        case param_s::YAW_FLTT:
+            return &attitude_control.get_rate_yaw_pid().filt_T_hz();
+        case param_s::YAW_FLTD:
+            return &attitude_control.get_rate_yaw_pid().filt_D_hz();
+        case param_s::YAW_FLTE:
+            return &attitude_control.get_rate_yaw_pid().filt_E_hz();
+        case param_s::YAW_FF:
+            return &attitude_control.get_rate_roll_pid().ff();
+        default:
+            GCS_SEND_TEXT(MAV_SEVERITY_EMERGENCY, "INTERNAL ERROR - get_param_pointer - param was %d", uint8_t(param));
+            // INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+            return nullptr;
+    }
+}
+
+float AP_Quicktune::get_param_value(AP_Quicktune::param_s param){
```

Comment:
```suggestion
float AP_Quicktune::get_param_value(AP_Quicktune::param_s param)
{
```

maybe run this file through our automatic `astyle`